e2e/Makefile

```diff
@@ -26,8 +32,82 @@ check-token:
 	  exit 1; \
 	fi
 
-install-terraform:
-	sudo apt-get install wget unzip
-	wget https://releases.hashicorp.com/terraform/0.11.13/terraform_0.11.13_linux_amd64.zip
-	unzip terraform_0.11.13_linux_amd64.zip
-	sudo mv terraform /usr/local/bin/
+#####################################################################
+# OS / ARCH
+#####################################################################
+OS=$(shell uname -s | tr '[:upper:]' '[:lower:]')
+ARCH=$(shell uname -m)
+ARCH_SHORT=$(ARCH)
+ifeq ($(ARCH_SHORT),x86_64)
+ARCH_SHORT := amd64
+else ifeq ($(ARCH_SHORT),aarch64)
+ARCH_SHORT := arm64
+endif
+
+## --------------------------------------
+## Build Dependencies
+## --------------------------------------
+
+##@ Build Dependencies:
+
+## Location to install dependencies to
+
+# Use CACHE_BIN for tools that cannot use devbox and LOCALBIN for tools that can use either method
+CACHE_BIN ?= $(CURDIR)/../bin
+LOCALBIN ?= $(CACHE_BIN)
+
+DEVBOX_BIN ?= $(DEVBOX_PACKAGES_DIR)/bin
+
+# if the $DEVBOX_PACKAGES_DIR env variable exists that means we are within a devbox shell and can safely
+# use devbox's bin for our tools
+ifdef DEVBOX_PACKAGES_DIR
+	LOCALBIN = $(DEVBOX_BIN)
+endif
+
+export PATH := $(CACHE_BIN):$(PATH)
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+##@ Tooling Binaries:
+GINKGO         ?= $(LOCALBIN)/ginkgo
+CTLPTL         ?= $(LOCALBIN)/ctlptl
+CLUSTERCTL     ?= $(LOCALBIN)/clusterctl
+KIND           ?= $(LOCALBIN)/kind
+KUSTOMIZE      ?= $(LOCALBIN)/kustomize
+
+## Tool Versions
+GINKGO_VERSION           ?= v2.13.2
+CTLPTL_VERSION           ?= v0.8.28
+CLUSTERCTL_VERSION       ?= v1.6.3
+KIND_VERSION             ?= v0.22.0
+KUSTOMIZE_VERSION        ?= v5.3.0
+
+.PHONY: tools
+tools: ginkgo ctlptl clusterctl kind kustomize
+
+.PHONY: ginkgo
+ginkgo: $(GINKGO)
+$(GINKGO): $(LOCALBIN) ## Download ginkgo locally if necessary.
+	GOBIN=$(LOCALBIN) go install github.com/onsi/ginkgo/v2/ginkgo@$(GINKGO_VERSION)
+
+.PHONY: ctlptl
+ctlptl: $(CTLPTL) ## Download ctlptl locally if necessary.
+$(CTLPTL): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install github.com/tilt-dev/ctlptl/cmd/ctlptl@$(CTLPTL_VERSION)
+
+.PHONY: clusterctl
+clusterctl: $(CLUSTERCTL) ## Download clusterctl locally if necessary.
+$(CLUSTERCTL): $(LOCALBIN)
+	curl -fsSL https://github.com/kubernetes-sigs/cluster-api/releases/download/$(CLUSTERCTL_VERSION)/clusterctl-$(OS)-$(ARCH_SHORT) -o $(CLUSTERCTL)
+	chmod +x $(CLUSTERCTL)
+
+.PHONY: kind
+kind: $(KIND) ## Download kustomize locally if necessary.
+$(KIND): $(LOCALBIN)
+	curl -fsSL https://github.com/kubernetes-sigs/kind/releases/download/v0.22.0/kind-$(OS)-$(ARCH_SHORT) -o $(KIND)
```

Comment:
```suggestion
	curl -fsSL https://github.com/kubernetes-sigs/kind/releases/download/$(KIND_VERSION)/kind-$(OS)-$(ARCH_SHORT) -o $(KIND)
```

Reply:
Ah yes absolutely! Must have missed that.